modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessScriptEngine.java

```diff
@@ -434,14 +397,35 @@ public Object run() {
             }, COMPILATION_CONTEXT);
         // Note that it is safe to catch any of the following errors since Painless is stateless.
         } catch (OutOfMemoryError | StackOverflowError | VerifyError | Exception e) {
-            throw convertToScriptException(scriptName == null ? source : scriptName, source, e);
+            throw convertToScriptException(source, e);
         }
     }
 
     void compile(Compiler compiler, Loader loader, MainMethodReserved reserved,
                  String scriptName, String source, Map<String, String> params) {
         final CompilerSettings compilerSettings;
 
+        compilerSettings = buildCompilerSettings(params);
```

Comment:
This can be one line like in the other compile method.

Reply:
@jdconrad thanks for the review, will clean that up before merging :)